compiler-core/src/type_/environment.rs

```diff
@@ -583,4 +632,60 @@ impl<'a, 'b> Environment<'a, 'b> {
             .map(|t| t.to_string())
             .collect()
     }
+
+    pub fn exhaustive(
+        &mut self,
+        patterns: Vec<Pattern<PatternConstructor, Arc<Type>>>,
+        value_typ: Arc<Type>,
+    ) -> bool {
+        // println!("patterns: {:?}", patterns);
+        // println!("value_typ: {:?}", value_typ);
+        match &*value_typ {
+            Type::App {
+                name: type_name,
+                module: module_vec,
+                ..
+            } => {
+                // TODO_EXH_CHECK this is bad.
+                // But I would prefer to not need to keep track of imported type names to imported constructor names.
+                // (because the imports can be named and usage can be a mix of qualified and imported constructor and type names)
```

Comment:
What does this mean?

Reply:
Right, I didn't word this clearly. When implementing this I needed to add a mapping from type to a list of its constructor names to the environment. I observed that the Environment and existing get_type / get_value functions work for local (potentially imported+aliased) names. But to keep things simple, and because in typed expressions types are referred to with original name, I decided not to import this mapping when importing a module. So this mapping is kept in its original module, and I need to reconstruct that module's name to a format that can be found in "importable_modules" in the environment (this is the "bad feeling" part, I didn't like that I needed to do this).

Maybe when printing better error messages, to print the patterns that user missed, this will need to be revisited and reconsidered.